non-oss/pharos_pro/commands/license_assign_command.rb

```diff
@@ -27,8 +29,108 @@ def execute
 
       Dir.chdir(config_yaml.dirname) do
         master_host.transport.connect
-        master_host.transport.exec!("kubectl create secret generic pharos-cluster --namespace=kube-system --from-literal='license.jwt=#{jwt_token.token}' --dry-run -o yaml | kubectl apply -f -")
-        logger.info "Added subscription token to pharos cluster secrets"
+        master_host.transport.exec!("kubectl create secret generic pharos-license --namespace=kube-system --from-literal='license.jwt=#{jwt_token.token}' --dry-run -o yaml | kubectl apply -f -")
+        logger.info "Assigned the subscription token successfully to the cluster.".green
+      end
+    end
+
+    private
+
+    def default_license_key
+      if tty?
+        prompt.ask('Enter Kontena Pharos license key:')
+      elsif !stdin_eof?
+        $stdin.read
+      else
+        signal_usage_error 'LICENSE_KEY required'
+      end
+    end
+
+    def jwt_token
+      @jwt_token ||= if license_key.match?(/^\h{8}-(?:\h{4}-){3}\h{12}$/)
+                       Pharos::LicenseKey.new(subscription_token, cluster_id: cluster_id)
+                     else
+                       begin
+                         load_config
+                         c_id = cluster_id
+                       rescue StandardError
+                         c_id = nil
+                       end
+                       Pharos::LicenseKey.new(license_key, cluster_id: c_id)
+                     end
+    end
+
+    def master_host
+      @master_host ||= load_config.master_host
+    end
+
+    def cluster_info
+      @cluster_info ||= Pharos::YamlFile.new(
+        StringIO.new(cluster_info_configmap),
+        override_filename: "#{master_host.address}:kube-public/cluster-info"
+      ).load
+    end
+
+    def cluster_info_configmap
+      Dir.chdir(config_yaml.dirname) do
+        master_host.transport.connect
+        master_host.transport.exec!('kubectl get configmap --namespace kube-public -o yaml cluster-info')
+      end
+    rescue Pharos::ExecError => ex
+      signal_error "Failed to get cluster-info configmap: #{ex.message}"
+    end
+
+    def cluster_id
+      cluster_info.dig('metadata', 'uid')
+    end
+
+    def cluster_name
+      load_config.name
```

Comment:
If no name exists, this is going to generate a new name and use it for now and it will be forgotten after that because it would be saved in the StoreClusterConfig phase when doing `up`.

Reply:
Though, you won't be doing `license assign` unless you have `up`ped already. Old clusters won't have a name (but they don't support the jwt either).

Reply:
New clusters will always have a name.